src/tests/test_analysis_vscode.py

```diff
@@ -1,51 +1,27 @@
-def test_analysis_eap(setup_vscode):
-    """
-    Test to run an analysis on migration from weblogic to EAP 7 in VSCode IDE
-    source : windup-rulesets/target/classes/eap7/weblogic/tests/data
-    target : eap7
-    """
-    vscode, config = setup_vscode
-    vscode.open_mta_perspective()
-    vscode.run_simple_analysis(
-        project=config["project_paths"]["eap7_generic"],
-        migration_target="eap7",
-    )
-    assert vscode.is_analysis_complete()
-    vscode.open_report_page()
-    assert vscode.verify_story_points(target="eap7")
-
-
-def test_analysis_eapxp(setup_vscode):
-    """
-    Test to run an analysis on migration from thorntail to eapxp2 in VSCode IDE
-    source : windup-rulesets/target/classes/eapxp/thorntail/tests/data
-    target : eapxp
-    """
-    vscode, config = setup_vscode
-    vscode.open_mta_perspective()
-    vscode.run_simple_analysis(
-        project=config["project_paths"]["eapxp_ruleset"],
-        migration_target="eapxp",
-    )
-    assert vscode.is_analysis_complete()
-    vscode.open_report_page()
-    assert vscode.verify_story_points(target="eapxp")
-
+import pytest
 
-def test_analysis_quarkus(setup_vscode):
-    """
-    Test to run an analysis on migration from quarkus in VSCode IDE. Note that quarkus1 was removed
-    as source and target through https://issues.redhat.com/browse/WINDUP-3328 .
 
-    source : windup-rulesets/target/classes/quarkus/springboot/tests/data
-    target : quarkus
-    """
+@pytest.mark.parametrize(
+    "migration_path",
+    ["weblogic_to_eap7", "thorntail_to_eapxp2", "springboot_to_quarkus", "eap_to_azure-appservice"],
+)
+def test_analysis(setup_vscode, migration_path):
+    """Analysis tests for VScode using various migration paths"""
     vscode, config = setup_vscode
     vscode.open_mta_perspective()
-    vscode.run_simple_analysis(
-        project=config["project_paths"]["quarkus_ruleset"],
-        migration_target="quarkus",
-    )
+    if migration_path == "weblogic_to_eap7":
```

Comment:
we can name the migration paths in such a way that we do not have to do these if else .


Example:



    "migration_path",
    ["eap7_generic", "eapxp_ruleset"],
)
def test_analysis(setup_vscode, migration_path):
 
    project = config["project_paths"][migraton_path]

Reply:
Done